lib/ansible/module_utils/facts/sysctl.py

```diff
@@ -18,21 +18,43 @@
 
 import re
 
+from ansible.module_utils._text import to_text
+
 
 def get_sysctl(module, prefixes):
     sysctl_cmd = module.get_bin_path('sysctl')
     cmd = [sysctl_cmd]
     cmd.extend(prefixes)
 
-    rc, out, err = module.run_command(cmd)
-    if rc != 0:
-        return dict()
-
     sysctl = dict()
-    for line in out.splitlines():
-        if not line:
-            continue
-        (key, value) = re.split(r'\s?=\s?|: ', line, maxsplit=1)
-        sysctl[key] = value.strip()
+
+    try:
+        rc, out, err = module.run_command(cmd)
+    except (IOError, OSError) as e:
+        module.warn('Unable to read sysctl: %s' % to_text(e))
+        rc = 1
+
+    if rc == 0:
+        key = ''
+        value = ''
+        for line in out.splitlines():
+            if not line.strip():
+                continue
+
+            if line.startswith(' '):
+                # handle multiline values, they will not have a starting key
+                value += line
```

Comment:
Is it intentional that this will preserve the leading space(s) within the concatenated multi-line value?

Also, is there any documentation on the output format, particularly as it pertains to multi-line values?

Reply:
Output format varies by platform I have found, so probably not. And I don't see anything in `man sysctl` about it on a BSD system (which is where the multiline issue was originally found).

@bcoca wrote that parsing code (I rebased it into a single commit after making other changes), so he might be the better person to ask about the spaces. I guess the other option would be to preserve the output exactly (so add the `\n` back in). No real preference here.

Reply:
Added the `\n` back in.